### PR TITLE
[CIAPP] Document custom spans

### PR DESCRIPTION
**Pull request by edrevo**
### What does this PR do?
Add documentation for the custom span feature

### Preview

https://docs-staging.datadoghq.com/ximo/add-custom-spans-doc/continuous_integration/setup_pipelines/custom_spans/

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

TODOs:
- [x] Blur identifiers in attached image
- [x] Release Jenkins plugin
- [x] Release datadog-ci
- [x] Release CircleCI support

### CODE REVIEW

File: content/en/continuous_integration/setup_pipelines/customspans.md
Comment (cg14823) [inline content/en/continuous_integration/setup_pipelines/customspans.md:43]: The block above this one says:

>  To trace a command, run the following:
> ```
> datadog-ci trace --name <name> <command>
> ```

Should the examples also include `--name` for consistency?

File: config/_default/menus/menus.en.yaml
Comment (juan-fernandez) [inline config/_default/menus/menus.en.yaml:2010]: nit, but wouldn't this be more consistent? 
```suggestion
    url: continuous_integration/setup_pipelines/custom_spans/
```

**Review by cg14823 — COMMENTED** [review]
In the preview this page also shows for us-fed and us3 which we don't currently support I think for those pages we should show an appropriate message saying it isn't supported yet.

File: content/en/continuous_integration/setup_pipelines/customspans.md
Comment (edrevo) [inline content/en/continuous_integration/setup_pipelines/customspans.md:19]: We should remove this or wait until we merge CircleCI documentation before merging this.

File: content/en/continuous_integration/setup_pipelines/customspans.md
Comment (juan-fernandez) [inline content/en/continuous_integration/setup_pipelines/customspans.md:23]: I'd probably specify the version from which this is available (`>0.17.0`)

File: content/en/continuous_integration/setup_pipelines/customspans.md
Comment (juan-fernandez) [inline content/en/continuous_integration/setup_pipelines/customspans.md:61]: do we want to show this between `"` ? 
Plus, is this the kind of value we want to encourage the user to use? This goes to operation name, doesn't it? These values are usually more like `grpc.client` or `junit.test`. Not that I don't think it wouldn't fit, just pointing it out.

**Review by juan-fernandez — APPROVED** [review]
LGTM, only minor stuff

File: content/en/continuous_integration/setup_pipelines/customspans.md
Comment (fermayo) [inline content/en/continuous_integration/setup_pipelines/customspans.md:2]:
```suggestion
title: Adding custom spans to Pipeline Traces
```

File: content/en/continuous_integration/setup_pipelines/customspans.md
Comment (fermayo) [inline content/en/continuous_integration/setup_pipelines/customspans.md:42]:
```suggestion
  --name "Greet" \
```

File: content/en/continuous_integration/setup_pipelines/customspans.md
Comment (edrevo) [reply content/en/continuous_integration/setup_pipelines/customspans.md:61]: We are not using `"` in the junit page. I can add them in both places if you want.

This is not the operations name (which will be `<ciprovider>.custom`). This maps to the resource name.

**Review by cg14823 — APPROVED** [review]
👍

**Comment by fermayo** [comment]
@edrevo I think all dependencies are now met, correct? Is this unblocked?

**Comment by edrevo** [comment]
Yes, the PR is unblocked but it seems like the history of master was rewritten so this PR is a mess. Fixing it.

**Review by kayayarai — APPROVED** [review]

**Pull request merged**
